### PR TITLE
Add the function of generating correlated graph pairs.

### PR DESCRIPTION
**Pull request by SHAAAAN**
Add a document in the graspy/graspy/simulations folder to generate the correlated pair of graphs.

### CODE REVIEW

File: graspy/simulations/sims_sample_corr.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    G1: ndarray (n_vertices, n_vertices)
+        Binary matrix the same size as P representing a random graph,
+        generated by the function of sample_edges
+    G2: ndarray (n_vertices, n_vertices)
+        Binary matrix the same size as P representing a random graph,
+        based on correlation matrix with P and Rho
+
+    Examples
+    --------
+    >>> p = 0.5
```
[the remark this answers is not in the document]
Comment (SHAAAAN) [reply graspy/simulations/sims_sample_corr.py]: Got it.

**Comment by SHAAAAN** [comment]
I add that line and renew the example.

**Comment by bdpedigo** [comment]
Give the PR a descriptive name and answer all of the questions in the PR template, please

File: graspy/simulations/sims_sample_corr.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
+import numpy as np
+from graspy.simulations import sample_edges
+import copy
+import warnings
+
+def sample_corr(P, Rho, directed=False, loops=False):
+    """
+    Generate a correlated Erdos-Renyi graph G2 based on G1 graph
```
Comment (bdpedigo) [inline graspy/simulations/sims_sample_corr.py]: a pair of correlated graphs

Comment (bdpedigo) [inline graspy/simulations/sims_sample_corr.py]: I know the authors call this ER, but lets not specify that here, I think it's confusing

File: graspy/simulations/sims_sample_corr.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    with Bernoulli distribution.
+
+    Both G1 and G2 are binary matrices. 
+    If the value of one position in graph 1 matrix is 1, 
+    the probability of the same position of graph 2 to be 1 
+    follows the Bernoulli distribution of
+    (P[i][j])+Rho[i][j]*(1-P[i][j]).
```
Comment (bdpedigo) [inline graspy/simulations/sims_sample_corr.py]: not necessary to go into the implementation of how to make the graphs correlated

File: graspy/simulations/sims_sample_corr.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    If the value of the position of graph 1 is 0, 
+    the probability of the same position of graph 2 to be 1 
+    follows the Bernoulli distribution of
+    P1[i][j] = P[i][j]*(1-Rho[i][j]).
+
+    Parameters
+    ----------
+    P: np.ndarray, shape (n_vertices, n_vertices)
```
Comment (bdpedigo) [inline graspy/simulations/sims_sample_corr.py]: make this the same as whatever we say for `P` in `sample_edges`

File: graspy/simulations/sims_sample_corr.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        Matrix of probabilities (between 0 and 1) for a random graph
+    Rho: np.ndarray, shape (n_vertices, n_vertices)
+        Matrix whose elements are real numbers in the interval [0,1]
+        Another probability matrix to definite the correlation between graph1 and graph2
```
Comment (bdpedigo) [inline graspy/simulations/sims_sample_corr.py]: not a probability matrix

File: graspy/simulations/sims_sample_corr.py [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    ----------
+    .. [1] Vince Lyzinski, et al. "Seeded Graph Matching for Correlated Erd}os-Renyi Graphs", 
+        Journal of Machine Learning Research 15, 2014
+        
+    Returns
+    -------
+    G1: ndarray (n_vertices, n_vertices)
+        Binary matrix the same size as P representing a random graph,
+        generated by the function of sample_edges
+    G2: ndarray (n_vertices, n_vertices)
```
Comment (bdpedigo) [inline graspy/simulations/sims_sample_corr.py]: Though this is indeed the process for generating a pair of correlated graphs, there is really nothing different about G1 and G2

File: graspy/simulations/sims_sample_corr.py [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+          [[0. 1. 0. 0. 0.]
+           [1. 0. 1. 0. 1.]
+           [0. 1. 0. 1. 1.]
+           [0. 0. 1. 0. 1.]
+           [0. 1. 1. 1. 0.]])
+    """
+
+    n = np.size(P,1)
+    G1 = sample_edges(P, directed = False, loops = False)
+    origin_G1 = copy.deepcopy(G1)
```
Comment (bdpedigo) [inline graspy/simulations/sims_sample_corr.py]: Don't need this

File: graspy/simulations/sims_sample_corr.py [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+           [0. 0. 1. 1. 0.]]
+          [[0. 1. 0. 0. 0.]
+           [1. 0. 1. 0. 1.]
+           [0. 1. 0. 1. 1.]
+           [0. 0. 1. 0. 1.]
+           [0. 1. 1. 1. 0.]])
+    """
+
+    n = np.size(P,1)
+    G1 = sample_edges(P, directed = False, loops = False)
```
Comment (bdpedigo) [inline graspy/simulations/sims_sample_corr.py]: the kwargs to `sample_edges` here should reflect those that were input to this function

File: graspy/simulations/sims_sample_corr.py [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+    origin_G1 = copy.deepcopy(G1)
+    # prob1 = origin_G1.sum()/n**2
+
+    P1 = copy.deepcopy(P)
```
Comment (bdpedigo) [inline graspy/simulations/sims_sample_corr.py]: don't use deepcopy, can just do `P.copy()`

File: graspy/simulations/sims_sample_corr.py [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+    Rho = copy.deepcopy(Rho)
+    for i in range(n):
```
Comment (bdpedigo) [inline graspy/simulations/sims_sample_corr.py]: as I mentioned, I still don't believe you need to use nested forloops here

File: graspy/simulations/sims_sample_corr.py [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+        for j in range(n):
+            if G1[i][j] == 1:
+                P1[i][j] = P[i][j]+Rho[i][j]*(1-P[i][j])
+            else:
+                P1[i][j] = P[i][j]*(1-Rho[i][j])
+    # prob2 = P1.sum()/n**2
+    
+    G2 = sample_edges(P1, directed = False, loops = False)
```
Comment (bdpedigo) [inline graspy/simulations/sims_sample_corr.py]: again, the kwargs should reflect those of the function

File: graspy/simulations/sims_sample_corr.py [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+    G2 = G2 - np.diag(np.diag(G2))
```
Comment (bdpedigo) [inline graspy/simulations/sims_sample_corr.py]: says who? this is only if the user specified `loops=False`. But if you fix the above comment then you don't need to worry about this

File: graspy/simulations/sims_sample_corr.py [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+    --------
+    >>> np.random.seed(1)
+    >>> p = 0.5
+    >>> rho = 0.3
+    >>> P = p * np.ones((5,5))
+    >>> Rho = rho * np.ones((5,5))
+
+    To sample a correlated graph based on P and Rho matrices:
+
+    >>> sample_corr(P,Rho,directed = False, loops = False)
```
Comment (bdpedigo) [inline graspy/simulations/sims_sample_corr.py]: I like having an example!

File: graspy/simulations/sims_sample_corr.py [hunk 14]
```diff
@@ -0,0 +1,91 @@
+import numpy as np
+from graspy.simulations import sample_edges
+import copy
+import warnings
+
+def sample_corr(P, Rho, directed=False, loops=False):
```
Comment (bdpedigo) [inline graspy/simulations/sims_sample_corr.py]: I am wondering if we should call this `sample_edges_corr` for consistency with the other function, what do you think?

File: graspy/simulations/sims_sample_corr.py [hunk 15]
```diff
@@ -0,0 +1,91 @@
+import numpy as np
+from graspy.simulations import sample_edges
```
Comment (bdpedigo) [inline graspy/simulations/sims_sample_corr.py]: have you tried doing this import as `from .simulations import sample_edges`?

**Comment by bdpedigo** [comment]
Modify the `__init__.py` file in this directory so that this function gets imported properly. See the current file for examples of what this should look like

**Comment by bdpedigo** [comment]
If you have questions about any of my comments, let's talk tomorrow?

File: graspy/simulations/sims_sample_corr.py (hunk 14 above)
Comment (SHAAAAN) [reply graspy/simulations/sims_sample_corr.py]: I agree

File: graspy/simulations/sims_sample_corr.py (hunk 15 above)
Comment (SHAAAAN) [reply graspy/simulations/sims_sample_corr.py]: Yeah, I can change it back.

**Pull request closed without merging**